### ISSUE TITLE
FP2344 Suffix 'Flags' suffix should be permitted when the Enum has the [Flags] attribute

### ISSUE THREAD

**Issue by paulhickman-a365** [SonarSource/sonar-dotnet#5743]
### Description

Rule S2344 fires when an enum name ends with Enum or Flags as a code style warning.

I believe that this should not happen if the enum name in question ends with Flags and also has the [Flags] attribute. 

The Flags suffix communicates important information to the caller that isn't provided by intellisense, namely that they are permitted to combine multiple values with an | operator.

### Repro steps

````c#
enum FooFlags { A, B , C }   //Should fire rule

[Flags]
enum BarFlags { A=1, B=2 , C=4 }   //Should not fire rule

enum FooEnum { A, B , C }   //Should fire rule

[Flags]
enum BarEnum { A=1, B=2 , C=4 }   //Should fire rule
````

### Known workarounds

Rule can be suppressed but that also suppresses checks for 'Enum' on the end of the name, which I wish to leave enabled.

Rule can be suppressed in code for each use if permitted by Sonar configuration.

### Related information

SonarAnalzyer.CSharp 8.40.0.48530

**Comment by pavel-mikula-sonarsource** [comment]
Hi @paulhickman-a365, the rule says
>The information that an enumeration type is actually an enumeration or a set of flags should not be duplicated in its name.

While it's beneficial to mark such enum with `[Flags]` attribute, it's still duplicating the same information. This time even from explicit annotation.

**Comment by perlun** [comment]
I agree with you @paulhickman-a365, this rule is a bit too picky for my taste also. Perhaps splitting this rule into two (one for `Enum` suffix and one for `Flags` suffix) would be an idea? That way, you could leave the `Enum` one enabled and disable the `Flags` one if that suits your particular project best.

**Comment by pavel-mikula-sonarsource** [comment]
Hi, 

as we don't closely monitor closed issues, please start a discussion on our community forum: https://community.sonarsource.com/

For example, the splitting idea could lead to an interesting outcome.
